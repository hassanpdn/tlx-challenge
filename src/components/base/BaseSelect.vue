<template>
      <div>
            <label for="base-select">{{label}}</label>
            <select @change="selectOption" id="base-select" class="select">
                  <option value="null" selected>Select a group size</option>
                  <option v-for="(item, index) in list" :key="`option-${index}`" :value="item" class="select-dropdown__list-item">{{item}}</option>
            </select>
      </div>   
</template>

<script>
      export default {
            name: 'base-select',
            props: {
                  list: {
                        type: Array,
                        default: () => {[]}
                  },
                  label:{
                        type: String,
                        default: ''
                  }
            },
            methods: {
                  selectOption(e){
                        this.$emit('select', e.target.value)
                  }
            }
      }
</script>

<style>
      .select {
            font-size: 12px;
            border: none;
            background: var(--gray);
            border-radius: 5px;
            padding: 8px;
            color: var(--white);
            cursor: pointer;
      }
      .select:focus {
            border: 1px solid var(--light-gray);
      }
      .select:focus-visible {
            outline: none;
      }
</style>