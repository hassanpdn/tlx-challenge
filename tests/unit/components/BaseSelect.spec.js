import { mount } from '@vue/test-utils';
import BaseSelect from '@/components/base/BaseSelect';

describe('BaseSelect', () => {
      test("renders component", () => {
            const wrapper = mount(BaseSelect);
            expect(wrapper.element).toMatchSnapshot()
      })

      test('setSelected demo', async () => {
            const wrapper = mount(BaseSelect, {propsData: {list: [10, 20, 30, 40]}})
            const options = wrapper.find('select').findAll('option')

            await options.at(1).setSelected()

            expect(wrapper.find('option:checked').element.value).toBe('10')
      })
});
