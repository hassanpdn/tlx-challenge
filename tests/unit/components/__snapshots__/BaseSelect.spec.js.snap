// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSelect renders component 1`] = `
<div>
  <label
    for="base-select"
  >
    
  </label>
   
  <select
    class="select"
    id="base-select"
  >
    <option
      selected="selected"
      value="null"
    >
      Select a group size
    </option>
     
  </select>
</div>
`;
